torchao/_models/llama/generate.py

```diff
@@ -206,20 +205,26 @@ def main(
             assert groupsize in [32,64,128,256], f"int4wo groupsize needs to be one of [32,64,128,256] but got {groupsize}"
             quantize(model, int4wo(groupsize=groupsize))
         if "autoquant" == quantization:
-            model = autoquant(model)
+            model = autoquant(model, manual=True)
+
             generate(
                 model,
                 encode_tokens(tokenizer, prompt, bos=True, device=device),
-                2,
-                interactive=False
+                max_new_tokens,
+                interactive=False,
+                temperature=temperature,
+                top_k=top_k,
             )
+
+            # do autoquantization
+            model.do_autoquant()
```

Comment:
can you also comment on why does this have to autoquant in this way

Reply:
because we optimize for the shapes autoquant sees during shape calibration so we have to run the full generate loop, which means we need a way to manually end shape calibration and initialize benchmarking/quantization

Reply:
we need to do shape calibration with the actual shapes used for generate, so we setup autoquant, set it to wait until we manually end shape calibration, run the generate set to log the correct shapes, then do_autoquant to actually do the benchmarks with the shapes that we've logged